### PR TITLE
Add support for comments

### PR DESCRIPTION
**Pull request by vjeux**
This wires up the comment logic from prettier. This needs some modification inside of prettier itself, so it won't work as is.

- locStart/End is ast-specific, we need a way to set it up for each parser. In this case I reused on that already existed: https://github.com/prettier/prettier/blob/master/src/util.js#L260-L262

- getSortedChildNodes as well, need a way to traverse the ast: https://github.com/prettier/prettier/blob/master/src/comments.js#L38

```js
        (node.kind && node.kind !== "Comment") ||
        node.ast_type && node.ast_type !== "comment") // <-- added
    ) {
```

- printComments as well: https://github.com/prettier/prettier/blob/master/src/comments.js#L928-L929

```js
  switch (comment.type || comment.kind || comment.ast_type) {
    case "comment":
      return comment.value;
    case "Comment":
```

If we tweak those two small things in prettier itself, then it's able to print comments using the heuristic. @azz, do you think you'd be able to extract those out to be parser-specific?

### CODE REVIEW

File: tests/python_comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    d
+
+    # e
+
+    call(
+        # a
+    )
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+if a: # a
```
Comment (taion) [inline tests/python_comments/__snapshots__/jsfmt.spec.js.snap]: This isn't quite correct. PEP 8 calls for two spaces before inline comments.

Comment (vjeux) [reply tests/python_comments/__snapshots__/jsfmt.spec.js.snap]: This is an odd requirement :)

Comment (taion) [reply tests/python_comments/__snapshots__/jsfmt.spec.js.snap]: See https://www.python.org/dev/peps/pep-0008/#inline-comments

> Inline comments should be separated by at least two spaces from the statement.

Yes, it's sort of weird, but it's more or less universal in Python-land.

File: vendor/python/astexport.py [hunk 2]
```diff
@@ -6,8 +6,15 @@
 
 
 def export_json(atok, pretty_print=False):
+    dict = export_dict(atok)
+    dict['comments'] = [{
+            'ast_type': 'comment',
+            'value': token.string,
+            'start': token.startpos,
+            'end': token.endpos,
+        } for token in atok.tokens if token.type == 57]
```
Comment (taion) [inline vendor/python/astexport.py]: should be `tokenize.COMMENT`, which is 53 on py2

**Pull request merged**